golang/src/metric_registrar_examples/metric_registrar_examples_test.go

```diff
@@ -1,103 +1,115 @@
-        })
-    })
-
-    Context("custom metric endpoint", func() {
-        It("increments the custom metric", func() {
-            resp, err := http.Get("http://localhost:8080/custom_metric")
-            Expect(err).ToNot(HaveOccurred())
-
-            expectOk(resp)
-
-            promResp, err := http.Get("http://localhost:8080/metrics")
-            Expect(err).ToNot(HaveOccurred())
-
-            bodyBytes, err := ioutil.ReadAll(promResp.Body)
-            Expect(err).ToNot(HaveOccurred())
-
-            lines := strings.Split(string(bodyBytes), "\n")
-            Expect(lines).To(ContainElement("custom 1"))
-        })
-    })
+	var session *gexec.Session
+	BeforeEach(func() {
+		session = startApp()
+	})
+
+	AfterEach(func() {
+		session.Terminate()
+		Eventually(session.Exited).Should(BeClosed())
+	})
+
+	Context("simple endpoint", func() {
+		It("returns OK", func() {
+			resp, err := http.Get("http://localhost:8080/simple")
+			Expect(err).ToNot(HaveOccurred())
+
+			expectOk(resp)
+		})
+	})
+
+	Context("high latency endpoint", func() {
+		It("returns OK slowly", func() {
+			start := time.Now()
+			resp, err := http.Get("http://localhost:8080/high_latency")
+			Expect(err).ToNot(HaveOccurred())
+			duration := time.Now().Sub(start)
+
+			expectOk(resp)
+			Expect(duration).To(BeNumerically(">", 2*time.Second))
+		})
+	})
+
+	Context("custom metric endpoint", func() {
+		It("increments the custom metric", func() {
+			resp, err := http.Get("http://localhost:8080/custom_metric?inc=1")
+			Expect(err).ToNot(HaveOccurred())
+
+			expectOk(resp)
+
+			promResp, err := http.Get("http://localhost:8080/metrics")
+			Expect(err).ToNot(HaveOccurred())
+
+			bodyBytes, err := ioutil.ReadAll(promResp.Body)
+			Expect(err).ToNot(HaveOccurred())
+
+			lines := strings.Split(string(bodyBytes), "\n")
+			Expect(lines).To(ContainElement("custom 1"))
+
+			resp, err = http.Get("http://localhost:8080/custom_metric")
+			Expect(err).ToNot(HaveOccurred())
+
+			promResp, err = http.Get("http://localhost:8080/metrics")
+			Expect(err).ToNot(HaveOccurred())
+
+			bodyBytes, err = ioutil.ReadAll(promResp.Body)
+			Expect(err).ToNot(HaveOccurred())
+
+			lines = strings.Split(string(bodyBytes), "\n")
+			Expect(lines).To(ContainElement("custom 0"))
+		})
+	})
 })
 
 func expectOk(resp *http.Response) {
-    Expect(resp.StatusCode).To(Equal(http.StatusOK))
+	Expect(resp.StatusCode).To(Equal(http.StatusOK))
 
-    bodyBytes, err := ioutil.ReadAll(resp.Body)
-    Expect(err).ToNot(HaveOccurred())
-    Expect(bodyBytes).To(BeEquivalentTo("{}"))
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(bodyBytes).To(BeEquivalentTo("{}"))
 }
 
 func startApp() *gexec.Session {
-    commandPath, err := gexec.Build(os.Getenv("GOPATH")+"/src/metric_registrar_examples", "-race")
-    Expect(err).ToNot(HaveOccurred())
+	commandPath, err := gexec.Build("./main.go", "-race")
```

Comment:
Is there a reason to change this to a relative path?

Reply:
Unless you set the GOPATH, there is no GOPATH, so when you push the app it actually can't load the UI.

Reply:
Odd.K.